includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -776,19 +777,24 @@ protected function getArticlesDetails( $articleIds, $articleKeys = [], $width =
 		$collection = $this->appendMetadata( $collection );
 
 		$thumbnails = null;
-		//if strict return to original ids order
+		$collection = $this->preserveOriginalOrder( $articleIds, $collection );
```

Comment:
Not sure if I understand the quetion (which loop?), but hope that my second comments will clarify my approach.